ugc/serdes.hpp

```diff
@@ -177,22 +189,47 @@ class DeserializerV0
   Source & m_source;
 };
 
+// Version must be used only for testing.
+// We should use only the latest version in production.
 template <typename Sink, typename UGC>
-void Serialize(Sink & sink, UGC const & ugc)
+void Serialize(Sink & sink, UGC const & ugc, Version const version = Version::Latest)
 {
-  WriteToSink(sink, static_cast<uint8_t>(Version::Latest));
+  WriteToSink(sink, static_cast<uint8_t>(version));
   Serializer<Sink> ser(sink);
   ser(ugc);
 }
 
-template <typename Source, typename UGC>
+template <typename Source>
 void Deserialize(Source & source, UGC & ugc)
 {
   uint8_t version = 0;
   ReadPrimitiveFromSource(source, version);
+  if (version == static_cast<uint8_t>(Version::V0) || version == static_cast<uint8_t>(Version::V1))
+  {
+    DeserializerV0<Source> des(source);
```

Comment:
Am I right that ugc v1 === ugc v0? I mean, do we need to decouple ugc update versions from ugc versions? If yes, please, create two different enums for versions. If not, please, parse ugc v1. Feel free to use DeserializerV0 for that.

Reply:
Discussed out loud.

Reply:
OK